examples/multi_gpu/distributed_sampling_quiver.py

```diff
@@ -0,0 +1,142 @@
+                x = self.convs[i]((x, x_target), edge_index)
+                if i != self.num_layers - 1:
+                    x = F.relu(x)
+                xs.append(x.cpu())
+
+                pbar.update(batch_size)
+
+            x_all = torch.cat(xs, dim=0)
+
+        pbar.close()
+
+        return x_all
+
+
+def run(rank, world_size, dataset, quiver_feature, quiver_sampler):
+    os.environ['MASTER_ADDR'] = 'localhost'
+    os.environ['MASTER_PORT'] = '12355'
+    dist.init_process_group('nccl', rank=rank, world_size=world_size)
+    torch.cuda.set_device(rank)
+
+    data = dataset[0]
+    train_idx = data.train_mask.nonzero(as_tuple=False).view(-1)
+    train_idx = train_idx.split(train_idx.size(0) // world_size)[rank]
+
+    train_loader = torch.utils.data.DataLoader(train_idx, batch_size=1024,
+                                               shuffle=True, num_workers=0)
+
+    if rank == 0:
+        subgraph_loader = NeighborSampler(data.edge_index, node_idx=None,
+                                          sizes=[-1], batch_size=2048,
+                                          shuffle=False, num_workers=6)
+
+    torch.manual_seed(12345)
+    model = SAGE(dataset.num_features, 256, dataset.num_classes).to(rank)
+    model = DistributedDataParallel(model, device_ids=[rank])
+    optimizer = torch.optim.Adam(model.parameters(), lr=0.01)
+
+    y = data.y.to(rank)
+
+    for epoch in range(1, 21):
+        model.train()
+
+        for seeds in train_loader:
+            n_id, batch_size, adjs = quiver_sampler.sample(seeds)
+            adjs = [adj.to(rank) for adj in adjs]
+
+            optimizer.zero_grad()
+            out = model(quiver_feature[n_id].to(rank), adjs)
+            loss = F.nll_loss(out, y[n_id[:batch_size]])
+            loss.backward()
+            optimizer.step()
+
+        dist.barrier()
+
+        if rank == 0:
+            print(f'Epoch: {epoch:03d}, Loss: {loss:.4f}')
+
+        if rank == 0 and epoch % 5 == 0:  # We evaluate on a single GPU for now
+            model.eval()
+            with torch.no_grad():
+                out = model.module.inference(quiver_feature, rank,
+                                             subgraph_loader)
+            res = out.argmax(dim=-1) == data.y
+            acc1 = int(res[data.train_mask].sum()) / int(data.train_mask.sum())
+            acc2 = int(res[data.val_mask].sum()) / int(data.val_mask.sum())
+            acc3 = int(res[data.test_mask].sum()) / int(data.test_mask.sum())
+            print(f'Train: {acc1:.4f}, Val: {acc2:.4f}, Test: {acc3:.4f}')
+
+        dist.barrier()
+
+    dist.destroy_process_group()
+
+
+if __name__ == '__main__':
+    dataset = Reddit('../../data/Reddit')
+    data = dataset[0]
+    world_size = torch.cuda.device_count()
+    print('Let\'s use', world_size, 'GPUs!')
+
+    #####################################
+    # Create Quiver Sampler And Feature
+    #####################################
+    csr_topo = quiver.CSRTopo(data.edge_index)
```

Comment:
Can we add a link to Quiver documentation here?

Reply:
OK~

Reply:
The link is now added.